plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2456,10 +2501,15 @@ else if (isMetadataColumnId(columnHandle.getId())) {
         }
 
         if (newEnforcedConstraint.equals(table.getEnforcedPredicate())
-                && newUnenforcedConstraint.equals(table.getUnenforcedPredicate())) {
+                && newUnenforcedConstraint.equals(table.getUnenforcedPredicate())
+                && constraint.getPredicateColumns().isEmpty()) {
```

Comment:
We should be computing `newConstraintColumns` before this `if` and checking if it is equal to `table.getConstraintColumns()`